### ISSUE TITLE
Intermittent hang in FUSE test suites

### ISSUE THREAD

**Issue by petemoore** [spectrum4/spectrum4#10]
Sometimes fuse tests don't run, and the process hangs. This is typically resolved by rerunning tup. However,

1) The tests shouldn't hang, so we should find out what is going on. I _think_ the hang occurs on startup before the individual tests are run, rather than e.g. midway through a test. Need to validate this hypothesis.
2) There is a rerun mechanism built into fuse-tests.sh that is intended to workaround the issue, yet it doesn't work, for some unknown reason. It implements retries for the test suite, yet if one fails, they all seem to fail. This is curious, since rerunning `tup` normally fixes it. Perhaps a sleep is needed? In any case, would be better to fix the problem at source.

One reason I've not looked too deeply is that the problem may lie in fuse, tup, bash, or something outside of this code base, and that may be tricky to diagnose.

**Comment by petemoore** [comment]
This failure now looks like this:

```
*    0) src/spectrum128k/tests: ./fuse-tests.sh test_po_t_udg.f101
[1936](https://github.com/spectrum4/spectrum4/actions/runs/3376735928/jobs/5604817005#step:3:1937)
./fuse-tests.sh: line 52: kill: (129150) - No such process
```

Example [here](https://github.com/spectrum4/spectrum4/actions/runs/3376735928/jobs/5604817005#step:3:1941)

**Comment by petemoore** [comment]
Retries for when the process dies was added in 5e8c05749afd99d5a2a5a952fde1c3f46c4e93e8 which seems to have fixed things, although I still have no idea what causes the underlying intermittent issue. Perhaps a bug in fuse. Will leave this open for a while to see if it comes back.

**Comment by petemoore** [comment]
I've managed to grab some core dumps, and so far found two independent issues. Reported upstream in:

* https://sourceforge.net/p/fuse-emulator/bugs/504/
* https://sourceforge.net/p/fuse-emulator/bugs/505/

**Comment by petemoore** [comment]
Proposed a fix in https://sourceforge.net/p/fuse-emulator/fuse/merge-requests/29/

**Comment by petemoore** [comment]
Fixed in 67e19f2716c8c1829488f490f196b6643722424e.
